packages/syft/src/syft/service/api/api_service.py

```diff
@@ -221,6 +229,145 @@ def api_endpoints(
 
         return api_endpoint_view
 
+    @service_method(
+        path="api.call_in_jobs", name="call_in_jobs", roles=GUEST_ROLE_LEVEL
+    )
+    def call_in_jobs(
+        self,
+        context: AuthedServiceContext,
+        path: str,
+        *args: Any,
+        **kwargs: Any,
+    ) -> Any | SyftError:
+        """Call a Custom API Method in a Job"""
+        return self._call_in_jobs(context, "call", path, *args, **kwargs)
+
+    @service_method(
+        path="api.call_private_in_jobs",
+        name="call_private_in_jobs",
+        roles=GUEST_ROLE_LEVEL,
+    )
+    def call_private_in_jobs(
+        self,
+        context: AuthedServiceContext,
+        path: str,
+        *args: Any,
+        **kwargs: Any,
+    ) -> Any | SyftError:
+        """Call a Custom API Method in a Job"""
+        return self._call_in_jobs(context, "call_private", path, *args, **kwargs)
+
+    @service_method(
+        path="api.call_public_in_jobs",
+        name="call_public_in_jobs",
+        roles=GUEST_ROLE_LEVEL,
+    )
+    def call_public_in_jobs(
+        self,
+        context: AuthedServiceContext,
+        path: str,
+        *args: Any,
+        **kwargs: Any,
+    ) -> Any | SyftError:
+        """Call a Custom API Method in a Job"""
+        return self._call_in_jobs(context, "call_public", path, *args, **kwargs)
+
+    def _call_in_jobs(
+        self,
+        context: AuthedServiceContext,
+        method: str,
+        path: str,
+        *args: Any,
+        **kwargs: Any,
+    ) -> Any | SyftError:
+        custom_endpoint = self.get_code(
+            context=context,
+            endpoint_path=path,
+        )
+        if isinstance(custom_endpoint, SyftError):
+            return custom_endpoint
+
+        result = context.node.add_api_endpoint_execution_to_queue(
+            context.credentials,
+            method,
+            path,
+            worker_pool=custom_endpoint.worker_pool,
+            *args,
+            **kwargs,
+        )
+        if isinstance(result, SyftError):
+            return result
+        # relative
+        from ..job.job_stash import JobStatus
+
+        # So result is a Job object
+        job = result
+        job_service = context.node.get_service("jobservice")
+        job_id = job.id
+        # Question: For a small moment, when job status is updated, it doesn't return the job during the .get() as if
+        # it's not in the stash. Then afterwards if appears again. Is this a bug?
+        timeout = 30  # 10 is too slow on GitHub CI macos?
```

Comment:
How can we handle this? A fixed timeout on a job of unknown length probably won't work, but no timeout might also be a problem.

Reply:
Should we pass timeout as an argument to the api with a considerable default timeout of let's say 300sec ?

Reply:
[0c5a372](https://github.com/OpenMined/PySyft/pull/8693/commits/0c5a37220b600effed51e82de8ccb5390d915045)
I just added a new field to the TwinAPIEndpoint settings so you can set different timeout values for each endpoint. Now, it's up to the admin to tweak these values whenever needed to keep things running smoothly.

Reply:
Excellent! @IonesioJunior ready for merge?